scripts/globals/utils.lua

```diff
@@ -571,3 +571,25 @@ function utils.lateralTranslateWithOriginRotation(origin, translation)
 
     return new_coords
 end
+
+function utils.canUseAbility(mob)
+    local act = mob:getCurrentAction()
+    if
+        act == xi.act.MOBABILITY_START or
+        act == xi.act.MOBABILITY_USING or
+        act == xi.act.MOBABILITY_FINISH or
+        act == xi.act.MAGIC_START or
+        act == xi.act.MAGIC_CASTING or
+        mob:getStatusEffect(xi.effect.STUN) ~= nil or
+        mob:getStatusEffect(xi.effect.PETRIFICATION) ~= nil or
+        mob:getStatusEffect(xi.effect.TERROR) ~= nil or
+        mob:getStatusEffect(xi.effect.SLEEP_I) ~= nil or
+        mob:getStatusEffect(xi.effect.SLEEP_II) ~= nil or
+        mob:getStatusEffect(xi.effect.AMNESIA) ~= nil or
```

Comment:
For completeness, you will also want to check `xi.effect.IMPAIRMENT` in case someone decides to call this on a player (or someone who's making a module uses Impairment on a mob for some reason)